test/extended/cluster/cl.go

```diff
@@ -137,7 +138,9 @@ var _ = g.Describe("[Feature:Performance][Serial][Slow] Load cluster", func() {
 
 				// Create templates as defined
 				for _, template := range p.Templates {
-					err := CreateTemplates(oc, c, nsName, e2e.TestContext.Viper, template, tuning)
+					step := metrics.NewTemplateStepDuration(tuning.Templates.RateLimit.Delay, tuning.Templates.Stepping.Pause)
```

Comment:
You're passing the same base object twice to this function (and `NewPodStepDuration`), also if there is no tuning set defined (ie. mastervert) you'll get a nil pointer panic.

Reply:
Added non-nil checking.


> You're passing the same base object twice to this function (and `NewPodStepDuration`)
The `Type` field is different in the 2 functions.
Maybe I missed your point here.

Reply:
I don't know what 2 functions you're talking about, I mean the single function `NewTemplateStepDuration`?

Why don't you just pass `tuning.Templates` directly and then in the function take whatever you need from it (`tuning.Templates.RateLimit.Delay`, `tuning.Templates.Stepping.Pause`)?

Reply:
That would introduce the dependency of `metrics` pkg on `cluster` pkg, which forms a cyclic one.